doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,118 @@
+
+Another example is if you want to create your own HTTP transport adapter for some other C++ HTTP stack. In this case, you can build your own HTTP transport adapter and also the libcurl HTTP transport adapter and run tests to compare the correctness of both. 
+
+## Using the HTTP transport adapter
+The HTTP transport adapter is set up when a service SDK client is init. For example, when creating an Azure Storage SDK client. The HTTP transport adapter is part of the client options argument that can be used to init the client. See the next example.
+```cpp
+  /*
+  * Using the HTTP transport adapter.
+  */
+
+  // When no options are provided, the HTTP transport adapter is set to the default one.
+  auto defaultStorageClient = BlobServiceClient(url, credential);
+
+  // Use client options to set an HTTP transport adapter.
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = std::make_shared<Azure::Core::Http::CurlTransport>();
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+### Re-use the HTTP transport adapter 
+Note that the HTTP transport adapter is a `shared_ptr`. This is because you can re-use the same HTTP transport adapter for multiple clients. There are two ways of doing this. The first one is by using the same HTTP transport adapter to init a second client. The second one is when the client can create a new client directly witch inherits the same configuration. See the next example:
+
+```cpp
+  /* 
+  *  Option 1. Init a new client with the same HTTP transport adapter.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, options);
+  // The second client.
+  BlobClientOptions optionsB;
+  optionsB.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientB = BlobServiceClient(url, credential, options);
+
+  /* 
+  *  Option 2. Create new client from a parent client.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, options);
+  // Create clientB from clientA.
+  auto clientB = storageClientA.GetBlobContainerClient(containerName)
+```
+
+Note that the second option could be not supported for some Azure SDK clients if it doesn't make sense for the service behavior (like Key Vault client).
+
+The Azure SDK for C++ client `will hold an internal copy of the options` passed in the arguments. This means that the options object used to init the client can be disposed. However, since the HTTP transport adapter is a shared ptr, it won't be disposed until any client using it is disposed first.
```

Comment:
Repeating `Azure SDK for C++` as a prefix too much hurts the readability and understanding of the concept at hand. It's OK to use it in some places, but not everywhere.

```suggestion
Each client holds an internal copy of the options passed in the arguments. This means that the options object used to create the client can be disposed. However, since the HTTP transport adapter is a shared pointer, it won't be disposed until all clients using it are disposed first.
```

Reply:
Agree.
The only benefit I see is when people copy a direct link to some paragraph tittle in the middle of the page like:
https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/core/Azure.Core#configuring-service-clients-using-clientoptions

In those cases, it could be good when the `Azure SDK for XX` is mentioned on every tittle to improve cases like this.

But I don't really care too much about it anyway.  I am really happy we have @ahsonkhan  in the team to help to improve the docs redaction and quality!